src/main/java/com/klarna/flink/connectors/dynamodb/DynamoDBBatchProcessor.java

```diff
@@ -200,18 +193,19 @@ public void flush() {
     // currently protected to allow overriding for testing. should be extracted from this class.
     protected ListenableFuture<BatchResponse> batchWrite(final BatchRequest batchRequest) {
         return listeningExecutorService.submit(() -> {
-            final BatchWriteItemRequest batchWriteItemRequest = new BatchWriteItemRequest();
-            batchWriteItemRequest.withRequestItems(batchRequest.getBatch());
+            final BatchWriteItemRequest batchWriteItemRequest = BatchWriteItemRequest.builder()
+                    .requestItems(batchRequest.getBatch())
+                    .build();
             boolean retry = false;
             int retries = 0;
             Throwable t = null;
             while (!retry && retries < 3) {
                 t = null;
                 try {
-                    final BatchWriteItemResult batchWriteItemResult = amazonDynamoDB.batchWriteItem(batchWriteItemRequest);
-                    if (!batchWriteItemResult.getUnprocessedItems().isEmpty()) {
+                    final BatchWriteItemResponse batchWriteItemResponse = dynamoDbClient.batchWriteItem(batchWriteItemRequest);
+                    if (batchWriteItemResponse.hasUnprocessedItems()) {
                         retry = true;
-                        batchWriteItemRequest.withRequestItems(batchWriteItemResult.getUnprocessedItems());
+                        batchWriteItemRequest.toBuilder().requestItems(batchWriteItemResponse.unprocessedItems());
```

Comment:
does toBuilder() return a new instance? if yes, need to use the new Instance

Reply:
Yes, mistake on my part. Addressed in [3e86faa](https://github.com/klarna-incubator/flink-connector-dynamodb/pull/28/commits/3e86faa7b97f06d987cb16a838804b50d55036df).